src/models/db/accounts.js

```diff
@@ -30,7 +30,11 @@ function * _upsertAccount (account, options) {
     existingAccount.setData(account)
     yield existingAccount.save(options)
   } else {
-    yield Account.create(account, options)
+    if (!account.balance) {
+      yield Account.create(_.merge({}, account, {balance: 0}), options)
+    } else {
+      yield Account.create(account, options)
+    }
```

Comment:
The problem with trying to enforce a not null constraint is that knex tries to insert `null` as the `default` on bulk inserts, so the db throw an error. We could work around this, since the only place we do bulk inserts is tests, but I wanted to keep the code changes as minimal as possible. I do think that setting a `not null` constraint is definitely a good approach long term, but that change could go into `master`.